app/services/UIBuilder.scala

```diff
@@ -53,48 +53,8 @@ object UIBuilder {
       }
     )
 
-  def pages(stanzaPages: Seq[models.ocelot.Page]): Map[String, Page] =
-    stanzaPages.map(p => (p.url, fromStanzaPage(p))).toMap
-
-
-  def fromText(txt: Phrase): Seq[TextItem] = {
-
-    def fromPattern(pattern: Regex, text: String): (List[String], List[Match]) =
-      (pattern.split(text).toList, pattern.findAllMatchIn(text).toList)
-
-    def matchesToLinks(enM: List[Match], cyM: List[Match]): List[HyperLink] =
-      enM.zip(cyM).map{ t =>
-        val( en, cy) = t
-        HyperLink( en.group(2), Text(en.group(1), cy.group(1)), false)
-      }
-
-    def textToTexts(enT: List[String], cyT: List[String]): List[Text] =
-      enT.zip(cyT).map{ t =>
-        val (en, cy) = t
-        Text(en, cy)
-      }
-
-    @tailrec
-    def joinTextsAndLinks(txts: List[Text], links: List[HyperLink], acc: Seq[TextItem]): Seq[TextItem] =
-      (txts, links) match {
-        case (Nil, Nil) =>  acc
-        case (t :: txs, l :: lxs ) => joinTextsAndLinks(txs, lxs, (acc :+t) :+ l)
-        case (t, Nil) => acc ++ t
-        case (Nil, _) => acc
-      }
-
-    val (enTexts, enMatches) = fromPattern(urlHttpLinkRegex, txt.langs(0))
-    val (cyTexts, cyMatches) = fromPattern(urlHttpLinkRegex, txt.langs(1))
-
-    joinTextsAndLinks(textToTexts(enTexts, cyTexts),
-                      matchesToLinks(enMatches, cyMatches),
-                      Nil)
+  def pages(stanzaPages: Seq[models.ocelot.Page]): Map[String, Page] = {
+    val stanzaIdToUrlMap = stanzaPages.map(p => (p.id, p.url)).toMap
```

Comment:
This assumes that all of the stanza ids for links are actually the ids for value stanzas at the start of a page. However, this is never checked when building a page. I think we should check this when building the original page.